rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -307,40 +305,71 @@ def __init__(self, pipeline, tensor_dtype = types.FLOAT, size = -1, auto_reset =
         self.output_list = None
         self.labels_size = self.batch_size
         self.output_memory_type = self.loader._output_memory_type
+        self.last_batch_padded_size = b.getLastBatchPaddedSize(self.loader._handle)
+        self.last_batch_policy = self.loader._last_batch_policy
+        self.shard_size =  size
+        self.auto_reset = auto_reset
+        self.batch_count = 0
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        if self.loader.rocal_run() != 0:
+        rocal_run = self.loader.rocal_run()
+        if rocal_run != 0:
+            if self.auto_reset:
+                self.reset()
+            raise StopIteration
+        elif self.shard_size > 0 and self.batch_count >= self.shard_size :
+            if self.auto_reset:
+                self.reset()
             raise StopIteration
         else:
             self.output_tensor_list = self.loader.get_output_tensors()
-        # Output list used to store pipeline outputs - can support multiple augmentation outputs
+        self.batch_count += self.batch_size
+        self.last_batch_size = self.batch_size - b.getLastBatchPaddedSize(self.loader._handle)
         self.output_list = []
         for i in range(len(self.output_tensor_list)):
             dimensions = self.output_tensor_list[i].dimensions()
             self.num_roi_dims = self.output_tensor_list[i].roi_dims_size()
             self.num_of_rois = self.num_roi_dims * 2
             self.roi_array = np.zeros(self.batch_size * self.num_of_rois, dtype=np.int32)
             self.output_tensor_list[i].copy_roi(self.roi_array)
-            torch_dtype = self.output_tensor_list[i].dtype()
+            roi = self.roi_array.reshape(self.batch_size,self.num_of_rois)
+            max_x1 = np.max(roi[...,self.num_of_rois-2:self.num_of_rois-1])
+            max_y1 = np.max(roi[...,self.num_of_rois-1:self.num_of_rois])
             if self.device == "cpu":
-                output = torch.empty(dimensions, dtype=getattr(torch, torch_dtype))
+                torch_dtype = self.output_tensor_list[i].dtype()
+                if (max_x1 == 0 or max_y1 == 0):
+                    output = torch.empty(dimensions, dtype=getattr(torch, torch_dtype))
```

Comment:
Why don't we alter the dimensions initially with max_x1 and max_y1 and then retain the same as before?

Reply:
because the values of max_x1 and max_y1 will change for every iteration